templates/default/deploy-scripts/src/deployer.rs

```diff
@@ -0,0 +1,394 @@
+use anyhow::{bail, Context, Error};
+use concordium_rust_sdk::smart_contracts::types::DEFAULT_INVOKE_ENERGY;
+use concordium_rust_sdk::types::hashes::TransactionHash;
+use concordium_rust_sdk::{
+    common::types::TransactionTime,
+    id::types::AccountAddress,
+    smart_contracts::common::ModuleReference,
+    types::{
+        queries::AccountNonceResponse,
+        smart_contracts::{ContractContext, InvokeContractResult, WasmModule},
+        transactions::{
+            self,
+            send::{deploy_module, init_contract, GivenEnergy},
+            InitContractPayload, UpdateContractPayload,
+        },
+        AccountTransactionEffects, BlockItemSummary, BlockItemSummaryDetails, ContractAddress,
+        Energy, TransactionType, WalletAccount,
+    },
+    v2::{self, BlockIdentifier},
+};
+use std::path::Path;
+use std::sync::Arc;
+
+/// A struct containing connection and wallet information.
+#[derive(Debug)]
+pub struct Deployer {
+    /// The client to establish a connection to a Concordium node (V2 API).
+    pub client: v2::Client,
+    /// The account keys to be used for sending transactions.
+    pub key: Arc<WalletAccount>,
+}
+
+impl Deployer {
+    /// A function to create a new deployer instance from a network client and a path to the wallet.
+    pub fn new(client: v2::Client, wallet_account_file: &Path) -> Result<Deployer, Error> {
+        let key_data = WalletAccount::from_json_file(wallet_account_file)
+            .context("Unable to read wallet file.")?;
+
+        Ok(Deployer {
+            client,
+            key: key_data.into(),
+        })
+    }
+
+    /// A function to check if a module exists on the chain.
+    pub async fn module_exists(
+        &mut self,
+        module_reference: &ModuleReference,
+    ) -> Result<bool, Error> {
+        let module_src = self
+            .client
+            .get_module_source(module_reference, &BlockIdentifier::LastFinal)
+            .await;
+
+        match module_src {
+            Ok(_) => Ok(true),
+            Err(e) if e.is_not_found() => Ok(false),
+            Err(e) => Err(e.into()),
+        }
+    }
+
+    /// A function to deploy a wasm module on the chain.
+    ///
+    /// If successful, the transaction hash, the block item, and
+    /// the module reference are returned.
+    /// If the module already exists on
+    /// chain, this function returns the module reference of the already
+    /// deployed module (no transaction hash or the block item returned).
+    ///
+    /// An optional expiry time for the transaction
+    /// can be given. If `None` is provided, the local time + 300 seconds is
+    /// used as a default expiry time.
+    pub async fn deploy_wasm_module(
+        &mut self,
+        wasm_module: WasmModule,
+        expiry: Option<TransactionTime>,
+    ) -> Result<
+        (
+            Option<TransactionHash>,
+            Option<BlockItemSummary>,
+            ModuleReference,
+        ),
```

Comment:
It usually makes sense to add a struct when there is a triple.

Is there a reason not to do so here?